packages/vue/src/components/molecules/SfModal/SfModal.js

```diff
@@ -48,49 +52,38 @@ export default {
   },
   methods: {
     close() {
-      this.$emit("close");
-      // remove left code
-    },
-    keydownHandler(e) {
-      if (e.key === "Escape" || e.key === "Esc" || e.keyCode === 27) {
-        this.close();
-      }
-      if (this.visible) {
-        // TODO: this is an error right now. what shall it do?
-        //  document.addEventListener("keydown", keydownHandler);
-        this.$emit("close");
-      } else {
-        // TODO: this is an error right now. what shall it do?
-        //  document.removeEventListener("keydown", keydownHandler);
-        this.$emit("open");
-      }
+      this.$emit("toggle", false);
     },
     checkPersistence() {
-      if (this.persistent === false) {
+      if (!this.persistent) {
         this.close();
       }
     }
   },
   watch: {
     visible: {
-      handler: function(visibility) {
-        if (visibility && typeof window !== "undefined") {
-          document.body.style.setProperty(
-            "margin-right",
-            `${window.innerWidth - document.body.clientWidth}px`
-          );
-          document.addEventListener("keydown", this.keydownHandler);
+      handler: value => {
+        if (typeof window === "undefined") return;
+        if (value) {
           document.body.style.setProperty("overflow", "hidden");
-        }
-        if (!visibility && typeof window !== "undefined") {
-          document.body.style.removeProperty("margin-right");
+        } else {
           document.body.style.removeProperty("overflow");
-          document.removeEventListener("keydown", this.keydownHandler);
         }
       },
       immediate: true
     }
   },
+  created() {
+    const escapeHandler = e => {
+      if (e.key === "Escape" && this.visible) {
+        this.close();
+      }
+    };
+    document.addEventListener("keydown", escapeHandler);
+    this.$once("hook:destroyed", () => {
```

Comment:
@aniamusial you never destroy this component, and then you never `removeEventListener`

Reply:
@przspa I think user would destroy the component:

```vue
<SfModal v-model="isModalVisible" v-if="something"/>
```

Anyway, would be better to `removeEventListener` when visible is set to false, right?

Reply:
@leomp12 but `v-if` and `v-mode` they don't make sense together :) 
To visible/hide model you use `v-model` if you use `v-if` then transition was broken.
Probably we can move `addEventListener` and `removeEventListener` to `watch`/

Reply:
@przspa you're right :)
Agree that would be better to watch `visible` to `addEventListener`/`removeEventListener`, also call this handler on `mounted` hook, right @aniamusial ?

Reply:
Maybe we can removed `mounted` and `created` action and use just 
```watch:{
  visible:{
    handler: function(){}
    immediate: true
  }
}
```
and remember:
![image](https://user-images.githubusercontent.com/12138170/69254306-5c607480-0bb6-11ea-83f0-9ccef00dcbb0.png)

Reply:
Also agree to watch visible. Previous behavior was assuming that we use v-if/v-show to show/hide modal. Now we do this internally so it's perfectly fine to use watcher. Also we should adjust SfSidebar to have same behavior